mesonbuild/dependencies/dev.py

```diff
@@ -505,6 +506,59 @@ def get_methods():
         return [DependencyMethods.SYSTEM]
 
 
+class JDKSystemDependency(ExternalDependency):
+    def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
+        super().__init__('jdk', environment, kwargs)
+
+        m = self.env.machines[self.for_machine]
+
+        java_home = os.environ.get('JAVA_HOME')
+        if java_home is None:
+            mlog.debug('$JAVA_HOME is unset')
+            self.is_found = False
+            return
+
+        self.version = environment.detect_java_compiler(self.for_machine).version
```

Comment:
I need to look tomorrow, but there may already be an instance of the java compiler available to you here.

Reply:
Cool thanks.